lib/integrations/externalcloudaudit/bootstrap.go

```diff
@@ -0,0 +1,405 @@
+	}
+
+	return nil
+}
+
+// createTransientBucket is similar to createLTSBucket however object locking is not enabled and instead a lifecycle
+// policy is created that cleans up transient storage:
+// * Query results expire after 1 day
+// * Large payloads expire after 4 days
+// * DeleteMarkers, NonCurrentVersions and IncompleteMultipartUploads are also removed
+func createTransientBucket(ctx context.Context, clt createBucketClient, bucketName string, region string) error {
+	err := createBucket(ctx, clt, bucketName, region, false)
+	if err != nil {
+		if !trace.IsAlreadyExists(err) {
+			return trace.Wrap(err)
+		}
+	}
+
+	_, err = clt.PutBucketLifecycleConfiguration(ctx, &s3.PutBucketLifecycleConfigurationInput{
+		Bucket: &bucketName,
+		LifecycleConfiguration: &s3types.BucketLifecycleConfiguration{
+			Rules: []s3types.LifecycleRule{
+				{
+					Status: s3types.ExpirationStatusEnabled,
+					ID:     aws.String("ExpireLargePayloads"),
+					Expiration: &s3types.LifecycleExpiration{
+						Days: 4,
+					},
+					Filter: &s3types.LifecycleRuleFilterMemberPrefix{
+						Value: "/large_payloads",
+					},
+				},
+				{
+					Status: s3types.ExpirationStatusEnabled,
+					ID:     aws.String("ExpireQueryResults"),
+					Expiration: &s3types.LifecycleExpiration{
+						Days: 1,
+					},
+					Filter: &s3types.LifecycleRuleFilterMemberPrefix{
+						Value: "/query_results",
+					},
+				},
+				{
+					Status: s3types.ExpirationStatusEnabled,
+					ID:     aws.String("ExpireNonCurrentVersionsAndDeleteMarkers"),
+					NoncurrentVersionExpiration: &s3types.NoncurrentVersionExpiration{
+						NewerNoncurrentVersions: 0,
+						NoncurrentDays:          1,
+					},
+					AbortIncompleteMultipartUpload: &s3types.AbortIncompleteMultipartUpload{
+						DaysAfterInitiation: 7,
+					},
+					Expiration: &s3types.LifecycleExpiration{
+						ExpiredObjectDeleteMarker: true,
+					},
+					Filter: &s3types.LifecycleRuleFilterMemberPrefix{},
+				},
+			},
+		},
+	})
+	if err != nil {
+		return trace.Wrap(awsutil.ConvertS3Error(err))
+	}
+
+	return nil
+}
+
+func createBucket(ctx context.Context, clt createBucketClient, bucketName string, region string, objectLock bool) error {
+	_, err := clt.CreateBucket(ctx, &s3.CreateBucketInput{
+		Bucket: &bucketName,
+		CreateBucketConfiguration: &s3types.CreateBucketConfiguration{
+			LocationConstraint: s3types.BucketLocationConstraint(region),
+		},
+		ObjectLockEnabledForBucket: objectLock,
+		ACL:                        s3types.BucketCannedACLPrivate,
+		ObjectOwnership:            s3types.ObjectOwnershipBucketOwnerEnforced,
+	})
+	if err != nil {
+		return trace.Wrap(awsutil.ConvertS3Error(err))
+	}
+
+	_, err = clt.PutBucketVersioning(ctx, &s3.PutBucketVersioningInput{
+		Bucket: &bucketName,
+		VersioningConfiguration: &s3types.VersioningConfiguration{
+			Status: s3types.BucketVersioningStatusEnabled,
+		},
+	})
+	if err != nil {
+		return trace.Wrap(awsutil.ConvertS3Error(err))
+	}
+
+	return nil
```

Comment:
```suggestion
	return trace.Wrap(awsutil.ConvertS3Error(err), "setting versioning configuration on S3 bucket")
```

Reply:
Updated in [b0fb795](https://github.com/gravitational/teleport/pull/33272/commits/b0fb79583182349c6856b505c55d951798ecf349) to include suggestion in addition to improvements in another area.